PrismSampleApp/PrismSampleApp/ViewModels/MainPageViewModel.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Prism.Commands;
+using Prism.Mvvm;
+using Prism.Navigation;
+using Prism.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using Xamarin.Forms;
+
+namespace PrismSampleApp.ViewModels
+{
+    public class MainPageViewModel : BindableBase, IConfirmNavigationAsync, INavigationAware
+    {
+        private string _title;
+        public string Title
+        {
+            get
+            {
+                return _title;
+
+            }
+            set
+            {
+                SetProperty(ref _title, value);
+            }
+        }
+        public MainPageViewModel(INavigationService navigationService, IPageDialogService pageDialogService)
+        {
+            Title = "Main Page";
+            _navigationService = navigationService;
+            _dialogService = pageDialogService;
+            NavigateCommandA = new Command(ExecuteNavigateCommandA);
+            NavigateCommandB = new Command(ExecuteNavigateCommandB);
+            NavigateCommandC = new Command(ExecuteNavigateCommandC);
+            NavigateCommandD = new Command(ExecuteNavigateCommandD);
+        }
+        private readonly INavigationService _navigationService;
+        private readonly IPageDialogService _dialogService;
+
+        public ICommand NavigateCommandA { get; set; }
+        public ICommand NavigateCommandB { get; set; }
+        public ICommand NavigateCommandC{ get; set; }
```

Comment:
Name the commands properly with their actions.

Reply:
Command have been properly renamed